x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_logs/view_logs_button.tsx

```diff
@@ -44,21 +48,21 @@ export const ViewLogsButton: React.FunctionComponent<ViewLogsProps> = ({
     const defaultStartTime = getFormattedRange(oneDayAgo);
     const defaultEndTime = getFormattedRange(now);
 
-    return logsLocator.getRedirectUrl({
+    return discoverLogsLocator?.getRedirectUrl({
       time: endTime ? endTime : defaultEndTime,
       timeRange: {
-        startTime: startTime ? startTime : defaultStartTime,
-        endTime: endTime ? endTime : defaultEndTime,
+        from: getTimeRangeStartFromTime(startTime ? startTime : defaultStartTime),
+        to: getTimeRangeEndFromTime(endTime ? endTime : defaultEndTime),
       },
       filter: logStreamQuery,
```

Comment:
Transferring the query that the Fleet UI sets doesn't seem to work. I guess the `filter` locator param doesn't mean the same thing in Discover than in the Logs UI. For Discover these are the structured filter pills while `query` corresponds to the free-form query. Do we need a translation here?

Reply:
Good catch, this should be a `query` instead